addons/auth_signup/models/res_users.py

```diff
@@ -164,6 +164,8 @@ def reset_password(self, login):
 
     def action_reset_password(self):
         """ create signup token for each user, and send their signup url by email """
+        if not self.filtered(lambda user: user.active):
```

Comment:
This doesn't mean that there is an archived user, it means that there isn't an active user on your recordset, i.e. all the users are archived.